core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -62,6 +56,20 @@ public BlockBuilder getFieldBlockBuilder(int fieldIndex)
         return fieldBlockBuilders[fieldIndex];
     }
 
+    void resetToSpecificIndex(int rowIndex)
+    {
+        this.rowIndex = rowIndex;
+        this.currentFieldIndexToWrite = 0;
```

Comment:
no need to set `currentFieldIndexToWrite` or `fieldBlockBuilderReturned` in this method